### PR TITLE
chore: dont rely on slice labels

### PR DESCRIPTION
**Pull request by MichaHoffmann**
I wanted to proceed a bit on trying to convert thanos to build with stringlabels and prom-label-proxy is used as a library there. This would fix this build error for me!

### CODE REVIEW

**Review by SuperQ — APPROVED** [review]

**Comment by squat** [comment]
@MichaHoffmann can you give me a bit more context? How is this code affecting external behavior in the Thanos build?

**Comment by SuperQ** [comment]
@squat There is the new "stringlabels" code in Prometheus itself. When switching to this implementation, accessing the slice directly is no longer allowed.

**Comment by SuperQ** [comment]
Here's what you get if you build prom-label-proxy now:

```
$ go build --tags=stringlabels .
# github.com/prometheus-community/prom-label-proxy/injectproxy
injectproxy/rules.go:210:24: cannot range over rule.Labels() (value of type "github.com/prometheus/prometheus/model/labels".Labels)
injectproxy/rules.go:234:23: cannot range over alert.Labels (variable of type "github.com/prometheus/prometheus/model/labels".Labels)
```

With this change, the build is successful.

**Comment by MichaHoffmann** [comment]
> @MichaHoffmann can you give me a bit more context? How is this code affecting external behavior in the Thanos build?

What @SuperQ said! Just adding context that I want to convert Thanos to be compatible with the stringlabels tag and there are more build failures, this was just one of them and it seemed easy enough to fix.

File: injectproxy/rules.go [hunk 1]
```diff
@@ -207,11 +207,8 @@ func (r *routes) filterRules(lvalues []string, resp *apiResponse) (interface{},
 	for _, rg := range rgs.RuleGroups {
 		var rules []rule
 		for _, rule := range rg.Rules {
-			for _, lbl := range rule.Labels() {
-				if lbl.Name == r.label && slices.Contains(lvalues, lbl.Value) {
-					rules = append(rules, rule)
-					break
-				}
+			if rule.Labels().Has(r.label) && slices.Contains(lvalues, rule.Labels().Get(r.label)) {
```
Comment (bboreham) [inline injectproxy/rules.go:210]: This would be more efficient calling just `Get` and checking for an non-empty result.

**Review by bboreham — COMMENTED** [review]
I’m not a maintainer here but had a comment if efficiency is a concern.

File: injectproxy/rules.go (hunk 1 above)
Comment (MichaHoffmann) [reply injectproxy/rules.go:210]: That makes sense! Thank you

Comment (MichaHoffmann) [reply injectproxy/rules.go]: changed it!

**Review by squat — APPROVED** [review]
Thanks for the context @SuperQ @MichaHoffmann 🏄‍♀️🏄‍♀️🏄‍♀️

**Comment by bboreham** [comment]
And soon `-tags dedupelabels` - https://github.com/prometheus/prometheus/pull/12304

**Pull request merged**